### PR TITLE
Added citation.cff

### PR DESCRIPTION
**Pull request by jmafoster1**
I added a citation.cff file to make it easier to cite the repo in publications. If you could please add your ORCID ID if you have one and check I've not missed any authors off, that'd be much appreciated.

### CODE REVIEW

File: citation.cff [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  - family-names: Clark
+    given-names: Andrew
+  - family-names: Foster
+    given-names: Michael
+    orcid: https://orcid.org/0000-0001-8233-9873
+  - family-names: Somers
+    given-names: Richard
+    orcid: https://orcid.org/0000-0002-1101-9722
+  - family-names: Turner
+    given-names: Robert
```
Comment (bobturneruk) [inline citation.cff:13]:
```suggestion
    given-names: Robert
    orcid: https://orcid.org/0000-0002-1353-1404
```

File: citation.cff [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  - family-names: Walkinshaw
+    given-names: Neil
+  - family-names: Wild
+    given-names: Christopher
+title: "Causal Testing Framework"
+version: 3.1.0
```
Comment (bobturneruk) [inline citation.cff:19]: @cwild-UoS can this (and below) be automatically updated somehow?

**Review by bobturneruk — APPROVED** [review]
Looks good. Very much in favour of citing software directly, rather than related papers.

**Review by cwild-UoS — APPROVED** [review]

File: citation.cff (hunk 2 above)
Comment (cwild-UoS) [reply citation.cff:20]: After a fair bit of digging around I can't find any tools or workflows to currently do this. I've found a good discussion on this in [another project](https://github.com/xarray-contrib/cf-xarray/issues/398).

Comment (bobturneruk) [reply citation.cff:20]: It's tricky. Might need to be done by hand for now. Similar problem for ORDA releases.

**Comment by christopher-wild** [comment]
Decided to cite with Figshare, PR with instructions on how to cite to soon follow

**Pull request closed without merging**